### DISCUSSION TITLE
I need help! i'm trying to do my models but it won't run! please help!

### DISCUSSION THREAD

**Discussion by jacobosaa** [Anura3D/Anura3D_OpenSource#126] [Q&A] [answered]
Hello everybody! My name is Jacobo. I'm a civil engineering student currently working on my thesis about using the Material Point Method (MPM) to calibrate rheological parameters in thickened tailings, such as yield stress and viscosity. This involves comparing the movement of columns with different sizes and solid concentrations. I'm also comparing the Mohr-Coulomb Model (which Anura has already implemented) with the Bingham model.

The primary objective is to find a range of solid concentrations where the behavior of the columns, using both models, yields similar results.

When it comes to modeling a column using the Mohr-Coulomb model in Anura, applying fixities, and following the steps outlined in the tutorial manual, I have no issues. However, I've encountered challenges while attempting to use the Bingham model following the same procedure. Despite performing the necessary steps, such as loading and releasing the fixities, the column doesn't collapse as expected.

I've reviewed the fixities applied to the geometry and believe they are correct. However, I've noticed that the sources file within the program lacks an "A3DBingham.dll" file. During my research, I came across code with that name, but due to my limited coding skills, I'm uncertain whether it would address my problem or not.

I will be attaching some pictures to this post for your reference. I appreciate your understanding of my non-native English and hope that someone reading this post can provide assistance. Greetings from Chile!

**Comment by jacobosaa** [comment]
![fixities liquid](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/0545f4bb-d3e1-4dd4-ad6b-56d482a843dd)
![fixities solid](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/ef3be754-b074-4175-9a33-83110a7a7512)
![material](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/ff5ffcf1-233f-445c-ac8e-6a047c26959c)
![paraview final](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/9ed5bf21-6cec-44ca-8d05-80f966bcb215)
![paraview inicio](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/08a3a1b2-58aa-4a15-9a59-446a1ffa629d)

**Reply by jacobosaa** [reply]
As you can observe, the initial three images depict the material definition and the process of adding and removing fixities. The last two images showcase the Paraview representation of the initial step (Step 0) and the final step, which should ideally correspond to the collapse phase. It's worth noting that I made modifications to the CPS_003 file as guided by the tutorial, even though the instructions were tailored for the Mohr-Coulomb model. I presumed that these changes, such as adjusting the number of steps and ensuring quasistatic convergence, were applicable to the Bingham model as well.

At this point, it's uncertain whether the issue lies with the fixities, the CPS file modifications, or the absence of the A3DBingham.dll file. As I strive to complete my thesis and progress toward becoming an engineer, resolving this challenge is imperative.

Hoping the best for this, i send you the best of my regards.

Jacobo

**Reply by YS-jy** [reply, marked as answer]
I have some questions about Bingham that I would like to ask you. Could you please share your email address?

**Reply by jacobosaa** [reply]
Hi! yes, sure, my e-mail is jaco.saavedra@gmail.com.  Have a great day!

**Comment by ayerrocolom** [comment]
Hello Jacobo,
The fixities should be the same independently on the material model that you use. What version of Anura3D are you using? When using the Bingham model, do you let the code run until you get final equilibrium?

**Comment by jacobosaa** [comment]
Hello Alba!
I'm currently using the Anura3D 2019.2 version, which is the version that
my guide professor gave me. I've tried to download and compile the latest
version of the Anura3D, but due to the lack of my computing abilities, I
wasn't able to get the correct file. I'm trying to do that, but I'm kinda
hopeless.
I can't get to the final equilibrium because I'm not able to get the files
from GiD (14.0.1) in the first place. I'm attaching an image of the message
that GiD sends me at that time I try to generate the Anura3d files.
Also, I must thank you for your time, I've read your work for my thesis and
truly admire you. It's truly an honour that you replied to me.
I'll be trying to get the new anura version.

**Comment by jacobosaa** [comment]
Hi Alba!

Hoping that this e-mail finds you well, i'm sending this message to tell
you that I've already upgraded my Anura3D software to the latest version,
and the problem that I used to have, is in the past. BUT, when it comes to
generate the CPS and GOM files, as the process finishes well, this message
appears. Is it ok? can i continue to analize the results?

Hoping to read from you soon!
Greetings from Chile!

[image: image.png]

**Comment by jacobosaa** [comment]
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/ec7c52f4-7565-4afe-826e-8ddf6e07e937)

**Reply by MJiaHao** [reply]
Hello, I also experienced this problem when installing the latest version of anura3d and the CPS file with GID output is empty. Have you solved it now? Can you give me some advice? I'm also using the 2019 Visual Studio version with 2023 HPC and Basekit Intel tools, and GID 16.1.6.

**Reply by christinealfradique** [reply]
@jacobosaa @MJiaHao, 
Me too! If you have some advice, please contact me.

**Reply by Tinsdrengre** [reply]
wuuuuuuu,Me,too! If you have some advice, please also contact me

**Reply by aube260** [reply]
Was there a solution to this error? I have the same error occurring using the tutorial oedometer example.

**Reply by aube260** [reply]
For me, this error was caused by some issue with the pre-processing in GiD. It created empty GOM and CPS files, as well as omitted the creation of the .mat and .prb files.
Refer to [https://github.com/Anura3D/Anura3D_OpenSource/discussions/186](https://github.com/Anura3D/Anura3D_OpenSource/discussions/186) for detail.

**Comment by jacobosaa** [comment]
May I add, that when i'm trying to analize those files, at the calculation step, i get this message.

![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/9f24b355-e368-4cf5-9337-af479d830afc)

**Comment by jacobosaa** [comment]
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/da3cd199-472a-4bb6-9c32-ab4e31cc1d8f)

**Reply by sunsh1nebread** [reply]
Hello, Jacob. May I ask how you updated the Anura3D software? I want to use the 2021 version of the software, but I'm unable to download the old version of OpenAPI that is compatible with Visual Studio 2017。Hoping to read from you soon!

**Reply by jacobosaa** [reply]
Hello Sunshine!

At first, I started using the 2019.2 version of anura (which I can send you if you want) that my guide professor gave me, and then I updated to the 2023 version using the 2023 tutorial manual with a YouTube video that I found. Everything went good and I'm currently using that version. 
I'm using the 2019 Visual Studio version, with the 2023 HPC and Basekit intel tools, and GID 16.0.6. 
Let me know if you want the 2019.2 version, which uses GID 14.

Best Regards!

**Reply by sunsh1nebread** [reply]
Hello, Jacob. Thank you very much for your suggestion. I have successfully upgraded according to the video tutorial
